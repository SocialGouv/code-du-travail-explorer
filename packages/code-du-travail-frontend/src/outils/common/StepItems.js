import React from "react";
import styled from "styled-components";
import { theme } from "@cdt/ui";
/**
 * For now, only <Step /> are used but stepItems will
 * allows to navigate between form's steps
 */
export function StepItems({ activeIndex = 0, items = [] }) {
  return (
    <StepItemsContainer>
      {items.map((item, index) => (
        <Step
          key={item.name}
          label={item.label}
          index={index}
          activeIndex={activeIndex}
        />
      ))}
    </StepItemsContainer>
  );
}

function Step({ label, index, activeIndex, ...props }) {
  const isActive = activeIndex === index;
  return (
    <StepWrapper index={index} isActive={isActive} {...props}>
      <IndexCircle isActive={isActive}>{index + 1}</IndexCircle>
      {label}
    </StepWrapper>
  );
}

const { colors, spacing, fonts } = theme;

const StepItemsContainer = styled.div`
  border-bottom: 1px solid ${colors.grey};
  margin-top: ${spacing.interComponent};
  padding-bottom: ${spacing.interComponent};
  display: flex;
  flex-wrap: wrap;
`;

const StepWrapper = styled.span`
  display: flex;
  width: 8.4375rem; // 135px
  align-items: center;
  padding: 0 ${spacing.tiny};
  font-size: ${fonts.sizeSmall};
  text-align: left;
  color: ${props => (props.isActive ? colors.blue : colors.grey)};
  text-decoration: none;
  font-weight: 600;
  line-height: 1;
`;

const IndexCircle = styled.span`
  border-radius: 49%;
  flex-shrink: 0;
  display: inline-flex;
  align-items: center;
  justify-content: center;
  height: 1.6rem;
  width: 1.6rem;
  margin: ${spacing.xsmall} ${spacing.small};
  margin-left: 0;
  color: ${colors.white};
  background-color: ${props => (props.isActive ? colors.blue : colors.grey)};
`;
