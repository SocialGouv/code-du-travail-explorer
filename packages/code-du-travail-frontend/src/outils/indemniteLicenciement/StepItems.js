import React from "react";
import styled from "styled-components";
import { theme } from "@cdt/ui";
/**
 * For now, only <Step /> are used but stepItems will
 * allows to navigate between formâ€™s steps
 */
export function StepItems({ activeIndex = 0, items = [] }) {
  return (
    <StepItemsContainer>
      {items.map((item, index) => (
        <Step
          key={item.name}
          label={item.label}
          index={index}
          activeIndex={activeIndex}
        />
      ))}
    </StepItemsContainer>
  );
}

function Step({ label, index, activeIndex, ...props }) {
  const isActive = activeIndex === index;
  return (
    <StepBase index={index} isActive={isActive} {...props}>
      <IndexCircle isActive={isActive}>{index + 1}</IndexCircle>
      {label}
    </StepBase>
  );
}

const { colors, spacing, fonts } = theme;

const StepItemsContainer = styled.div`
  border-bottom: 1px solid ${colors.grey};
  padding: ${spacing.interComponent} 0;
  display: flex;
  flex-wrap: wrap;
`;

const StepBase = styled.span`
  text-align: left;
  flex-basis: 130px;
  display: flex;
  align-items: center;
  padding-left: ${props => (props.index === 0 ? 0 : spacing.interComponent)};
  text-decoration: none;
  font-size: ${fonts.sizeSmall};
  font-weight: 600;
  line-height: 1;
  color: ${props => (props.isActive ? colors.blue : colors.grey)};
`;

const IndexCircle = styled.span`
  border-radius: 49%;
  flex-shrink: 0;
  display: inline-flex;
  align-items: center;
  justify-content: center;
  height: 1.6rem;
  width: 1.6rem;
  margin: ${spacing.xsmall} ${spacing.small};
  margin-left: 0;
  color: ${colors.white};
  background-color: ${props => (props.isActive ? colors.blue : colors.grey)};
`;
