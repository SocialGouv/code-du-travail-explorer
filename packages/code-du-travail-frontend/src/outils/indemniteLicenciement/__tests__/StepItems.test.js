import React from "react";
import { render } from "react-testing-library";
import { StepItems } from "../StepItems";

describe("<StepItems />", () => {
  it("should render", () => {
    const items = [
      { name: "step-1", label: "Step One" },
      { name: "step-2", label: "Step Two" },
      { name: "step-3", label: "Step Three" }
    ];

    const { container } = render(<StepItems items={items} activeIndex={1} />);
    expect(container).toMatchSnapshot();
  });
});
