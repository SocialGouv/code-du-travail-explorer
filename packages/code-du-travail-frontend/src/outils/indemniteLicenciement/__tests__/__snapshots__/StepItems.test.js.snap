// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StepItems /> should render 1`] = `
.c0 {
  border-bottom: 1px solid #adb9c9;
  padding: 1.25rem 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c1 {
  text-align: left;
  -webkit-flex-basis: 130px;
  -ms-flex-preferred-size: 130px;
  flex-basis: 130px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-left: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 0.9rem;
  font-weight: 600;
  line-height: 1;
  color: #adb9c9;
}

.c3 {
  text-align: left;
  -webkit-flex-basis: 130px;
  -ms-flex-preferred-size: 130px;
  flex-basis: 130px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-left: 1.25rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 0.9rem;
  font-weight: 600;
  line-height: 1;
  color: #0053b3;
}

.c5 {
  text-align: left;
  -webkit-flex-basis: 130px;
  -ms-flex-preferred-size: 130px;
  flex-basis: 130px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-left: 1.25rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 0.9rem;
  font-weight: 600;
  line-height: 1;
  color: #adb9c9;
}

.c2 {
  border-radius: 49%;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 1.6rem;
  width: 1.6rem;
  margin: 0.5rem 0.625rem;
  margin-left: 0;
  color: #fff;
  background-color: #adb9c9;
}

.c4 {
  border-radius: 49%;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 1.6rem;
  width: 1.6rem;
  margin: 0.5rem 0.625rem;
  margin-left: 0;
  color: #fff;
  background-color: #0053b3;
}

<div>
  <div
    class="c0"
  >
    <span
      class="c1"
    >
      <span
        class="c2"
      >
        1
      </span>
      Step One
    </span>
    <span
      class="c3"
    >
      <span
        class="c4"
      >
        2
      </span>
      Step Two
    </span>
    <span
      class="c5"
    >
      <span
        class="c2"
      >
        3
      </span>
      Step Three
    </span>
  </div>
</div>
`;
